### PR TITLE
Allow duplicate refable node names across packages

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves: https://github.com/dbt-labs/dbt-core/issues/7446
resolves: https://github.com/dbt-labs/dbt-core/issues/1269

### Description
* removes DuplicateResourceNameError call in manifest parsing
* uses _find_unique_ids_for_package in RefableLookup
* raises AmbiguousResourceNameRefError in RefableLookup.get_unique_id if multiple unique_ids found
* plumbs caller node (source_node) through RefableLookup for node-aware CompilationError message

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

```
Compilation Error in model model_downstream (models/model_downstream.sql)
  dbt found multiple potential nodes when referencing "model" - ['model.project_b.model', 'model.project_a.model'].
  To fix this, specify which package to use as the first argument to 'ref' (one of: ['project_b', 'project_a']).
```

### CODE REVIEW

File: core/dbt/exceptions.py [hunk 1]
```diff
@@ -1961,6 +1961,21 @@ def get_message(self) -> str:
...
+    def __init__(self, duped_name, unique_ids, node=None):
+        self.duped_name = duped_name
+        self.unique_ids = unique_ids
+        self.packages = [unique_id.split(".")[1] for unique_id in unique_ids]
+        super().__init__(msg=self.get_message(), node=node)
+
+    def get_message(self) -> str:
+        msg = (
+            f'dbt found multiple potential nodes when referencing "{self.duped_name}" - {self.unique_ids}.'
+            f"\nTo fix this, specify which package to use as the first argument to 'ref' (one of: {self.packages})."
```
Comment (jtcohen6) [inline core/dbt/exceptions.py:1974]: This is great! Teensy nits around the formatting of our lists:
```
Compilation Error in model model_downstream (models/model_downstream.sql)
  dbt found multiple potential nodes when referencing 'model': 'model.project_b.model', 'model.project_a.model'
  To fix this, use two-argument 'ref' with the package name ('project_b' or 'project_a')
```

I think a join on `"or"` is fine even if there's >2:
```
('project_a' or 'project_b' or 'project_c' ...)
```

**Review by jtcohen6 — COMMENTED** [review]
nice!! took it for a spin, & it spins

shall we queue up https://github.com/dbt-labs/dbt-core/issues/1269 to close? :)

File: core/dbt/exceptions.py (hunk 1 above)
Comment (MichelleArk) [reply core/dbt/exceptions.py]: Thank you, updated!

**Comment by jtcohen6** [comment]
Ah, something that's still not working: attaching yaml properties to the correct node in the case of duplicates.

```yml
# packages.yml
packages:
  - local: package_a
  - local: package_b
```
```sql
-- package_a/models/my_model.sql
select 1 as id
```
```yml
# package_a/models/schema.yml
models:
  - name: my_model
    description: My cool model
```
```sql
-- package_b/models/my_model.sql
{{ config(alias="something_else") }}
select 1 as id
```
```
-- models/another_model.sql
select * from {{ ref('package_a', 'my_model') }}
```
```
$ dbt run
06:40:32  Running with dbt=1.6.0-a1
...
Compilation Error
  When referencing 'my_model', dbt found nodes in multiple packages: 'model.package_b.my_model', 'model.package_a.my_model'
  To fix this, use two-argument 'ref', with the package name first: 'package_b' or 'package_a'
```

**Goal:** It should be possible to define yaml properties for a model within the same package as that model.

(In the future, we could look to support ["overrides"](https://docs.getdbt.com/reference/resource-properties/overrides) of the package resource from the root project, similar to how we do with `sources`. That's definitely out of scope for now.)

**Comment by MichelleArk** [comment]
@jtcohen6 - Pushed up a fix for defining yaml properties on a model within the same package as that model.

**Pull request merged**